packages/block-library/src/gallery/style.scss

```diff
@@ -1,131 +1,180 @@
-	list-style-type: none;
-	padding: 0;
-	// Some themes give all <ul> default margin instead of padding.
-	margin: 0;
-
-	.blocks-gallery-image,
-	.blocks-gallery-item {
+	// Need bogus :not(#individual-image) to override long :not()
+	// specificity chain on default image block on front end.
+	figure.wp-block-image:not(#individual-image) {
 		// Add space between thumbnails, and unset right most thumbnails later.
-		margin: 0 1em 1em 0;
-		display: flex;
-		flex-grow: 1;
-		flex-direction: column;
-		justify-content: center;
-		position: relative;
-		align-self: flex-start;
+		margin: 0 var(--gallery-block--gutter-size, #{$grid-unit-20}) var(--gallery-block--gutter-size, #{$grid-unit-20}) 0;
+
+		&:last-of-type:not(#individual-image) {
+			margin-right: 0;
+		}
 
-		// On mobile and responsive viewports, we allow only 1 or 2 columns at the most.
-		width: calc(50% - 1em);
+		width: calc(50% - (var(--gallery-block--gutter-size, #{$grid-unit-20}) / 2));
 
 		&:nth-of-type(even) {
 			margin-right: 0;
 		}
+	}
+
+	figure.wp-block-image {
+		display: flex;
+		flex-grow: 1;
+		justify-content: center;
+		position: relative;
+		margin-top: auto;
+		margin-bottom: auto;
+		// IE11 doesn't like the "flex-direction: column;" here.
+		@supports ( position: sticky ) {
+			flex-direction: column;
+		}
 
-		figure {
+		> div,
+		> a {
 			margin: 0;
-			height: 100%;
 
-			// IE doesn't support flex so omit that.
-			@supports (position: sticky) {
-				display: flex;
-				align-items: flex-end;
-				justify-content: flex-start;
+			// Avoid applying flex styles to IE11.
+			@supports ( position: sticky ) {
+				flex-direction: column;
+				flex-grow: 1;
 			}
 		}
 
 		img {
 			display: block;
-			max-width: 100%;
 			height: auto;
-
-			// IE doesn't handle cropping, so we need an explicit width here.
+			max-width: 100%;
 			width: 100%;
 
+			// IE doesn't handle cropping, so we need an explicit width here.
 			// IE11 doesn't read rules inside this query. They are applied only to modern browsers.
-			@supports (position: sticky) {
+			@supports ( position: sticky ) {
 				width: auto;
 			}
 		}
 
 		figcaption {
-			position: absolute;
+			background: linear-gradient(0deg, rgba($color: $black, $alpha: 0.7) 0, rgba($color: $black, $alpha: 0.3) 70%, transparent);
 			bottom: 0;
-			width: 100%;
+			color: $white;
+			font-size: $default-font-size;
+			left: 0;
+			margin-bottom: 0;
 			max-height: 100%;
 			overflow: auto;
-			padding: 3em 0.77em 0.7em;
-			color: $white;
+			padding: 40px 10px 9px;
+			position: absolute;
 			text-align: center;
-			font-size: 0.8em;
-			background: linear-gradient(0deg, rgba($color: $black, $alpha: 0.7) 0, rgba($color: $black, $alpha: 0.3) 70%, transparent);
-			box-sizing: border-box;
-			margin: 0;
+			width: 100%;
 
 			img {
 				display: inline;
 			}
 		}
+
+		&.is-style-rounded {
+			> div,
+			> a {
+				// Not supported in IE11.
+				@supports ( position: sticky ) {
+					flex: 1 1 auto;
+				}
```

Comment:
I have [added an issue for this](https://github.com/WordPress/gutenberg/issues/33483) in the follow up after merge column on the [gallery refactor project board](https://github.com/WordPress/gutenberg/projects/56). Given this will be deleted I won't worry about making the formatting change at this point.